engine/src/main/java/org/camunda/bpm/engine/impl/util/JsonUtil.java

```diff
@@ -726,6 +727,7 @@ public static Gson getGsonMapper() {
   public static Gson createGsonMapper() {
     return new GsonBuilder()
       .serializeNulls()
+      .setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
```

Comment:
I don't think it is redundant because without this line we would use a different date format but you are right, after I changed the `TelemetryReporterTest` this is not needed anymore. However, I think it would still make sense to use our default date format instead of the default format of Gson. Unfortunately, I found no easy way to pull in the configured date format for the REST API here.
In the end, this Gson formatter is currently only used to send the telemetry data to our backend so leaving it as is, is probably fine. We could also argue that changing it would not be in the scope of this feature.

I will remove it. :)